docs/howtos/linkerd2.md

```diff
@@ -167,6 +167,146 @@ Congratulations! You're successfully routing traffic to the QOTM application, th
+  acmeProvider:
+    authority: none
+  requestPolicy:
+    insecure:
+      action: Route
+```
+
+2. Configure the target cluster Ambassador to support Linkerd health checks.
+
+Multicluster Linkerd does its own health checks beyond what Kubernetes does, so a `Mapping` is needed to allow Linkerd's health checks to succeed:
+
+```yaml
+apiVersion: getambassador.io/v2
+kind: Mapping
+metadata:
+  name: public-health-check
+  namespace: ambassador
+spec:
+  prefix: /-/ambassador/ready
+  rewrite: /ambassador/v0/check_ready
+  service: localhost:8877
+  bypass_auth: true
+```
+
+When configuring Ambassador, Kubernetes is usually configured to run health checks directly against port 8877 -- however, that port is not meant to be exposed outside the cluster. The `Mapping` permits accessing the health check endpoint without directly exposing the port.
+
+(The actual prefix in the `Mapping` is not terribly important, but it needs to match the metadata supplied to the service mirror controller, below.)
+
+3. Configure the target cluster Ambassador for the service mirror controller.
+
+This requires changes to the Ambassador's `deployment` and `service`. **For all of these commands, you will need to make sure your Kubernetes context is set to talk to the target cluster.**
+
+- In the `deployment`, you need the `config.linkerd.io/enable-gateway` `annotation`:
+
+```bash
+$ kubectl -n ambassador patch deploy ambassador -p='
+spec:
+    template:
+        metadata:
+            annotations:
+                config.linkerd.io/enable-gateway: "true"
+'
+```
+
+- In the `service`, you need to provide appropriate named `port` definitions:
+
+   - `mc-gateway` needs to be defined as `port` 4143
+   - `mc-probe` needs to be defined as `port` 80, `targetPort` 8080 (or wherever Ambassador is listening)
+
+```bash
+$ kubectl -n ambassador patch svc ambassador --type='json' -p='[
+        {"op":"add","path":"/spec/ports/-", "value":{"name": "mc-gateway", "port": 4143}},
+        {"op":"replace","path":"/spec/ports/0", "value":{"name": "mc-probe", "port": 80, "targetPort": 8080}}
+    ]'
+```
+
+- Finally, the `service` also needs its own set of `annotation`s:
+
+```bash
+$ kubectl -n ambassador patch svc ambassador -p='
+metadata:
+    annotations:
+        mirror.linkerd.io/gateway-identity: ambassador.ambassador.serviceaccount.identity.linkerd.cluster.local
+        mirror.linkerd.io/multicluster-gateway: "true"
+        mirror.linkerd.io/probe-path: -/ambassador/ready
+        mirror.linkerd.io/probe-period: "3"
+'
+```
+
+(Here, the value of `mirror.linkerd.io/probe-path` must match the `prefix` using for the probe `Mapping` above.)
+
+4. Configure individual exported services. Adding the following annotations to a service will tell the service to use Ambassador as the gateway:
+
+```bash
+$ kubectl -n $namespace patch svc $service -p='
+metadata:
+    annotations:
+        mirror.linkerd.io/gateway-name: ambassador
+        mirror.linkerd.io/gateway-ns: 
+'
```

Comment:
shouldn't this also have `ambassador` for the `gateway-ns`?

Reply:
Yes indeed -- fixed, thanks!